docs/operations-guide/running-metabase-on-heroku.md

```diff
@@ -40,41 +40,101 @@ Now that you’ve installed Metabase, it’s time to [set it up and connect it t
     * The most likely culprit here is a stale database migrations lock that was not cleared.  This can happen if for some reason Heroku kills your Metabase dyno at the wrong time during startup.  __To fix it:__ you can either clear the lock using the built-in [release-locks](../troubleshooting-guide/application-database.md) command line function, or if needed you can login to your Metabase application database directly and delete the row in the `DATABASECHANGELOGLOCK` table.  Then just restart Metabase.
 
 
-# Deploying New Versions of Metabase
+## Deploying New Versions of Metabase
 
-Upgrading to the next version of Metabase is a simple process where you will grab the latest version of [metabase-deploy](https://github.com/metabase/metabase-deploy) and push it to Heroku.
+We currently use a Heroku buildpack for deploying Metabase. The [metabase-deploy](https://github.com/metabase/metabase-deploy) repository is an app that relies on the buildpack and configures various properties for running Metabase on Heroku.
+
+In order to upgrade to the latest version of Metabase on Heroku, you need to trigger a rebuild of your app. Typically this is done by pushing to the Heroku app's Git repository, but because we create the app using a Heroku Button, you'll need to link your app to our repo and push an empty commit.
 
 Here's each step:
 
-* Clone the latest version to your local machine:
+* Clone the metabase-deploy repo to your local machine:
 
 ```bash
 git clone https://github.com/metabase/metabase-deploy.git  
 cd metabase-deploy
 ```
 
-* Add a git remote with your metabase setup:
+* Add a git remote with your Metabase setup:
 
 ```bash
-git remote add heroku https://git.heroku.com/your-metabase-app.git
+git remote add heroku https://git.heroku.com/<YOUR-APP-NAME>.git
 ```
 
 * If you are upgrading from a version that is lower than 0.25, add the Metabase buildpack to your Heroku app:
-```
+
+```bash
 heroku buildpacks:add https://github.com/metabase/metabase-buildpack
 ```
 
-* Force push the new version to Heroku:
+* If there have been no new changes to the `metabase-deploy` repository since the last time you deployed Metabase, you will need to add an empty commit. This triggers Heroku to re-deploy the code, fetching the newest version of Metabase in the process.
```

Comment:
I know this is how things actually work, but this seems really lame as an upgrade flow =(

Reply:
Agreed. I hunted around for a bit to see if there was a better way, but didn’t come across much. This is part of the reason I included the pipelines component - it at least provides a testable and continuous deployment mechanism. Something I would want were I running Metabase in production on Heroku. That said, it still requires the empty commit strategy.